lib/directory-tree.js

```diff
@@ -27,16 +27,18 @@ function directoryTree (path, options, onEachFile) {
 
 	try { stats = FS.statSync(path); }
 	catch (e) { return null; }
-	if (stats.isFile()) {
-		const ext = PATH.extname(path).toLowerCase();
 
-		// Only add files with the provided extensions
-		if (options && 
-			options.extensions && 
-			options.extensions.length && 
-			options.extensions.indexOf(ext) === -1)
-			return null;
+	// Skip if it matches the exclude regex
+	if (options && options.exclude && options.exclude.test(path))
+		return null;  
 
+	if (stats.isFile()) {
+		// Skip if it does not match the include regex
+		if (options && options.includeOnly && !options.includeOnly.test(path))
```

Comment:
Sounds good. I completely agree, it's better to get it right than to do it fast.
The exception with the `exclude` param is that it doesn't matter at what level the file/folder is excluded, an entire matching subdirectory can be ignored without any side effects. 
However, with the `include` it needs to test till the last child, which is where the cyclic situation arises. How about something called `includeFileRegex` or `includeFileOnlyIf`.

We are yet again faced with one of the most difficult things in programming i.e. naming variables. 😋 
Let me know if you can think of a better param name.